Changing permissions not permitted

I'm having an issue running the CI and here is the log.

```
Run php-actions/phpstan@v3
  with:
    path: src
    configuration: tools/phpstan/phpstsan.neon
    version: composer
    php_version: latest
    command: analyse
Run set -e
  set -e
  bash <(curl -s https://raw.githubusercontent.com/php-actions/php-build/59e915e4b8eeab830231a5dce6ad0998adebac05/php-build.bash) phpstan
  /home/runner/work/_actions/php-actions/phpstan/v3/phpstan-action.bash
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    ACTION_TOKEN: ***
    ACTION_VERSION: composer
    ACTION_PHP_VERSION: latest
    ACTION_PHPSTAN_PATH: 
    ACTION_PHP_EXTENSIONS: 
    ACTION_COMMAND: analyse
    ACTION_PATH: src
    ACTION_CONFIGURATION: tools/phpstan/phpstsan.neon
    ACTION_LEVEL: 
    ACTION_PATHS_FILE: 
    ACTION_AUTOLOAD_FILE: 
    ACTION_ERROR_FORMAT: 
    ACTION_GENERATE_BASELINE: 
    ACTION_MEMORY_LIMIT: 
    ACTION_ARGS: 
Building PHP latest with extensions:  ...
chmod: changing permissions of '/home/runner/work/[PROJECT]/[PROJECT]/vendor/bin/phpstan': Operation not permitted
Error: Process completed with exit code 1.
```

CI was working well 3 days ago and today this issue. No change to the CI configuration of the project were made.

It seems that some new commits appends to this repos, maybe it could be related.

--- a/phpstan-action.bash
+++ b/phpstan-action.bash
@@ -25,7 +25,11 @@ else
 	phar_path="${GITHUB_WORKSPACE}/$ACTION_PHPSTAN_PATH"
 fi
 
-chmod +x "$phar_path"
+if [ ! -x "$phar_path" ];
+then
+	chmod +x "$phar_path"
+fi
+
 command_string=("phpstan")
 
 if [ -n "$ACTION_COMMAND" ]

